src/urllib3/util/retry.py

```
@@ -335,6 +338,8 @@ def get_retry_after(self, response: BaseHTTPResponse) -> float | None:
     def sleep_for_retry(self, response: BaseHTTPResponse) -> bool:
         retry_after = self.get_retry_after(response)
         if retry_after:
+            if self.max_retry_wait_length and retry_after > self.max_retry_wait_length:
+                raise MaxRetryWaitExceededError(retry_after_header_value=retry_after)
```

The `is_retry` method probably should be updated too to respect the `max_retry_wait_length`.
e.g. the [responses library relies on it.](https://github.com/getsentry/responses/blob/489b84768af3be97a2f6df6fabcc0dd8d749e2a9/responses/__init__.py#L1154)

---

Why? A response can still be a retry response even if its `Retry-After` header exceeds the limit, right? I don't see why we should raise an error just for checking this.

---

A better method name for what you want might be something like `is_retriable()`.